tests/connection.rs

```diff
@@ -202,3 +202,65 @@ fn server_inactive() {
 
     assert_eq!(app.world().resource::<ServerTick>().get(), 0);
 }
+
+#[test]
+fn after_deferred_replication_start() {
```

Comment:
Let's use `is_empty` instead of comparing actual IDs. Yes, it's more robust to compare IDs, but I think that in this case it will be more clear if just validate if a client were added to the resource by using the mentioned method.

Also with this suggestion it will look nicer if we create a variable for resource instead of putting everything into the assert.